chatmodel-springai/src/test/java/com/example/ai/controller/ChatControllerTest.java

```diff
@@ -76,6 +78,26 @@ void sentimentAnalyzer() {
                 .body("answer", is("SARCASTIC"));
     }
 
+    @Test
+    void embeddingClientConversion() {
+        String response = given().contentType(ContentType.JSON)
+                .body(new AIChatRequest("Why did the Python programmer go broke? Because he couldn't C#"))
+                .when()
+                .post("/api/ai/embedding-client-conversion")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .contentType(ContentType.JSON)
+                .extract()
+                .jsonPath()
+                .get("answer");
+
+        double[] doubles = Arrays.stream(response.replaceAll("[\\[\\]]", "").split(","))
+                .mapToDouble(Double::parseDouble)
+                .toArray();
+
+        assertThat(doubles.length).isEqualTo(1536).as("Dimensions for openai model is 1536");
```

Comment:
@coderabbitai, there is no method called allSatisfy in assertj library, please fix below suggestion

```java
assertThat(doubles).allSatisfy(value ->
           assertThat(value).isBetween(-1.0, 1.0)
        ).as("All embedding values should be between -1 and 1");
```